### PR TITLE
4) rust -os flag

### PR DESCRIPTION
**Pull request by crop2000**

### CODE REVIEW

**Comment by crop2000** [comment]
based on top of #1077 #1080 #1072

**Comment by crop2000** [comment]
@sletz finally this passes the tests too, so it is ready.

File: compiler/generator/instructions_compiler.cpp [hunk 1]
```diff
@@ -1138,9 +1138,14 @@ ValueInst* InstructionsCompiler::generateVariableStore(Tree sig, ValueInst* exp)
                     return IB::genLoadStructVar(vname);
                 }
             } else {
-                // "Slow" variables are declared as locals in 'frame' or 'compute' functions
-                pushComputeBlockMethod(IB::genDecStackVar(vname, ctype, exp));
-                return IB::genLoadStackVar(vname);
+                if (gGlobal->gOneSample) {
```
Comment (sletz) [inline compiler/generator/instructions_compiler.cpp:1141]: Why this change is needed ?

File: compiler/generator/instructions_compiler.cpp [hunk 2]
```diff
@@ -1178,9 +1183,13 @@ ValueInst* InstructionsCompiler::generateVariableStore(Tree sig, ValueInst* exp)
                             getConditionCode(sig), IB::genStoreStructVar(vname_perm, exp)));
                         return IB::genLoadStructVar(vname_perm);
                     } else {
-                        // copy the object variable to the local one
-                        pushComputeBlockMethod(
-                            IB::genDecStackVar(vname, ctype, IB::genLoadStructVar(vname_perm)));
+                        if (gGlobal->gOneSample) {
```
Comment (sletz) [inline compiler/generator/instructions_compiler.cpp:1186]: Why this change is needed ?

File: compiler/generator/rust/rust_code_container.cpp [hunk 3]
```diff
@@ -632,13 +636,56 @@ void RustCodeContainer::generateComputeHeader(int n, std::ostream* fOut, int fNu
...
           << "&mut self, " << fFullCount << ": usize, inputs: & [& [FaustFloat] ]"
           << ", outputs: & mut[& mut[FaustFloat] ]) {";
     tab(n + 1, *fOut);
 }
 
+void RustCodeContainer::generateComputeFrame(int n)
+{
+    tab(n, *fOut);
+    tab(n, *fOut);
+    *fOut << "pub fn compute_frame(&mut self, inputs: &[&FaustFloat], ";
```
Comment (sletz) [inline compiler/generator/rust/rust_code_container.cpp:649]: Can you use `frame` instead of `compute_frame` to be similar to what C and C++ already generate ?

File: compiler/generator/instructions_compiler.cpp (hunk 1 above)
Comment (sletz) [reply compiler/generator/instructions_compiler.cpp:1141]: This break the C/C++ backends. Compare the generated code in Rust and C/C++.

Comment (crop2000) [reply compiler/generator/instructions_compiler.cpp:1141]: when i -os and don't have -ec i use the generateControlDeclarations before the "loop" part. only when i put "slow" variable into the declare block they will be available. same for the other comment

Comment (sletz) [reply compiler/generator/instructions_compiler.cpp:1141]: Change in `Instructions_compiler.cpp` may break other backends which is the case here. So try to save the issue in the Rest backend itself and do not change `Instructions_compiler.cpp`.

File: compiler/generator/rust/rust_code_container.cpp (hunk 3 above)
Comment (crop2000) [reply compiler/generator/rust/rust_code_container.cpp:649]: i prefer compute_frame because it makes clearer that it is a variant of compute.

**Comment by crop2000** [comment]
@sletz i changed this pr and also tested cpp with -os in ci

**Comment by sletz** [comment]
Thanks, merged, with the "public" function renaming and different fix in the tests (make c/make cpp were broken).
